asynciterator.ts

```diff
@@ -1796,6 +1796,9 @@ export class ClonedIterator<T> extends TransformIterator<T> {
   constructor(source: AsyncIterator<T>) {
     super(source, { autoStart: false });
     this._reading = false;
+    // As cloned iterators are not auto-started, they must always be marked as readable.
+    if (source)
```

Comment:
I suspect that rather than *always* setting the iterator to be readable if there is a source, it will be sufficient to add `|| source.readable` to the [condition here](https://github.com/RubenVerborgh/AsyncIterator/blob/3171521759194b3ce463f989d8253e35978ef690/asynciterator.ts#L1836-L1837).

Reply:
I just tried, and it looks like this doesn't fix the two Comunica issues mentioned above anymore unfortunately.

Reply:
Hmm ok. I still don't like just setting `.readable` to be true here because we are glossing over some kind of race condition taking place in the iterator.

Is it possible that the problematic cases are occurring when we are calling `.clone()` on a source that has already ended. In that case it might be that we need to be checking `source.done` instead of `source.readable` in that conditional.

Reply:
> Is it possible that the problematic cases are occurring when we are calling .clone() on a source that has already ended. In that case it might be that we need to be checking source.done instead of source.readable in the conditional on line 1836.

Just tried this as well, not solving the problem either unfortunately.


Note that this change is essentially just restoring the behaviour of `BufferedIterator`, which is setting `readable` to `true` if autostart is disabled: https://github.com/RubenVerborgh/AsyncIterator/blob/3171521759194b3ce463f989d8253e35978ef690/asynciterator.ts#L971
We need to do this, since `ClonedIterator` is voiding the implementation of `BufferedIterator`'s `_init`.